libs/books/feature/src/lib/reading-list/reading-list.component.ts

```diff
@@ -1,22 +1,44 @@
-import { Component } from '@angular/core';
+import { Component, OnInit, OnDestroy } from '@angular/core';
 import { MatSnackBar } from '@angular/material/snack-bar';
 import { Store } from '@ngrx/store';
-import { getReadingList, removeFromReadingList, undoRemoveFromReadingList } from '@tmo/books/data-access';
+import { Actions, ofType } from '@ngrx/effects';
+import { Subscription } from 'rxjs';
+import {
+  getReadingList,
+  removeFromReadingList,
+  undoRemoveFromReadingList,
+  confirmedRemoveFromReadingList
+} from '@tmo/books/data-access';
 import { ReadingListItem } from '@tmo/shared/models';
 
 @Component({
   selector: 'tmo-reading-list',
   templateUrl: './reading-list.component.html',
   styleUrls: ['./reading-list.component.scss']
 })
-export class ReadingListComponent {
+export class ReadingListComponent implements OnInit, OnDestroy {
   readingList$ = this.store.select(getReadingList);
 
-  constructor(private readonly store: Store, private readonly snackBar: MatSnackBar) { }
+  subscription: Subscription;
+
+  constructor(
+    private readonly store: Store,
+    private readonly snackBar: MatSnackBar,
+    private readonly actions$: Actions
+  ) {}
+
+  ngOnInit(): void {
+    this.subscription = this.actions$
```

Comment:
Similarly here. I would never recommend you do this.

Reply:
Moved logic to effect class